core/src/main/java/bisq/core/dao/node/lite/network/RequestBlocksHandler.java

```diff
@@ -168,31 +167,36 @@ public void onFailure(@NotNull Throwable throwable) {
-                        checkArgument(connection.getPeersNodeAddressOptional().isPresent(),
-                                "RequestDataHandler.onMessage: connection.getPeersNodeAddressOptional() must be present " +
-                                        "at that moment");
-                        cleanup();
-                        log.info("We received from peer {} a BlocksResponse with {} blocks",
-                                nodeAddress.getFullAddress(), getBlocksResponse.getBlocks().size());
-                        listener.onComplete(getBlocksResponse);
-                    } else {
-                        log.warn("Nonce not matching. That can happen rarely if we get a response after a canceled " +
-                                        "handshake (timeout causes connection close but peer might have sent a msg before " +
-                                        "connection was closed).\n\t" +
-                                        "We drop that message. nonce={} / requestNonce={}",
-                                nonce, getBlocksResponse.getRequestNonce());
-                    }
-                } else {
-                    log.warn("We have stopped already. We ignore that onDataRequest call.");
-                }
-            } else {
-                log.warn("We got a message from ourselves. That should never happen.");
+            if (stopped) {
+                log.warn("We have stopped already. We ignore that onDataRequest call.");
+                return;
+            }
+
+            Optional<NodeAddress> optionalNodeAddress = connection.getPeersNodeAddressOptional();
+            if (!optionalNodeAddress.isPresent()) {
+                log.warn("Peers node address is not present, that is not expected.");
+                // We do not return here as in case the connection has been created from the peers side we might not
+                // have the address set. As we check the nonce later we do not care that much for the check if the
+                // connection address is the same as the one we used.
+            } else if (!optionalNodeAddress.get().equals(nodeAddress)) {
+                log.warn("Peers node address is the same we requested. We ignore that message.");
+                return;
             }
+
+            GetBlocksResponse getBlocksResponse = (GetBlocksResponse) networkEnvelope;
+            if (getBlocksResponse.getRequestNonce() != nonce) {
+                log.warn("Nonce not matching. That can happen rarely if we get a response after a canceled " +
+                                "handshake (timeout causes connection close but peer might have sent a msg before " +
+                                "connection was closed).\n\t" +
+                                "We drop that message. nonce={} / requestNonce={}",
+                        nonce, getBlocksResponse.getRequestNonce());
+                return;
+            }
+
+            cleanup();
```

Comment:
Doesn't timeoutTimer need to be stopped here like before?
```
        stopTimeoutTimer();
```

Reply:
cleanup calls stopTimeoutTimer. But will consolidate cancel/stop/cleanup